DashKit/DashPumpManager.swift

```diff
@@ -668,43 +702,44 @@ public class DashPumpManager: PumpManager {
-                        completion(.success(dose))
-                    case .failure(let error):
-                        self.mutateState({ (state) in
-                            state.activeTransition = nil
-                        })
-                        self.finalizeAndStoreDoses()
-                        completion(.failure(DashPumpManagerError(error)))
+                // SDK not allowing us to make calls from a callback thread, so dispatch.
+                DispatchQueue.global(qos: .userInteractive).async {
+                    self.podCommManager.sendProgram(programType: program, beepOption: .init(beepAtEnd: false)) { (result) in
+                        switch result {
+                        case .success(let podStatus):
+                            self.mutateState({ (state) in
+                                state.unfinalizedTempBasal = UnfinalizedDose(tempBasalRate: enactRate, startTime: startDate, duration: duration, scheduledCertainty: .certain)
+                                state.updateFromPodStatus(status: podStatus)
+                                state.activeTransition = nil
+                            })
+                            self.finalizeAndStoreDoses()
+                            completion(.success(dose))
+                        case .failure(let error):
+                            self.mutateState({ (state) in
+                                state.activeTransition = nil
+                            })
+                            self.finalizeAndStoreDoses()
+                            completion(.failure(DashPumpManagerError(error)))
+                        }
                     }
                 }
             }
         }
     }
 
     public func setMustProvideBLEHeartbeat(_ mustProvideBLEHeartbeat: Bool) {
-        // Seems to be causing a deadlock in the SDK
-//        if mustProvideBLEHeartbeat {
-//            podCommManager.configPeriodicStatusCheck(interval: .minutes(1)) { (result) in
-//                switch result {
-//                case .failure(let error):
-//                    self.log.error("podCommManager periodic status check error: %{public}@", String(describing: error))
-//                case .success(let status):
-//                    self.log.debug("podCommManager periodic status: %@", String(describing: status))
-//                    self.pumpDelegate.notify({ (delegate) in
-//                        delegate?.pumpManagerBLEHeartbeatDidFire(self)
-//                    })
-//                }
-//            }
-//        }
+        if mustProvideBLEHeartbeat && !isPeriodicStatusCheckConfigured {
+            self.log.debug("podCommManager periodic status: configuring")
+            podCommManager.configPeriodicStatusCheck(interval: .minutes(1)) { (result) in
+                switch result {
+                case .failure(let error):
+                    self.log.error("podCommManager periodic status check error: %{public}@", String(describing: error))
```

Comment:
If this happens (and other sources of BLE heartbeat) fail then Loop won't loop. Should we check and warn about this condition? Perhaps at app level if there are no working BLE heartbeat devices? Or is the chance of that so remote?

Reply:
A couple things: this configuration (pump driving the heartbeat) is primarily used for testing; the CGM should generally drive things.  We *do* enable it if we don't hear from the CGM in a while, but by that point we can't loop as we do not have fresh cgm data.  Secondly, we do attempt to assert that his is enabled several times, like on *any* cgm update, including errors, or when the app is made active.